gui/storage/forms.py

```diff
@@ -1080,15 +1093,23 @@ def clean_disk_passwd2(self):
 
     def clean(self):
         cdata = self.cleaned_data
-        if not cdata.get("disk_passwd"):
+        if cdata.get('disk_reset_password'):
+            cdata['disk_passwd'] = ''
+        elif not cdata.get("disk_passwd"):
             cdata['disk_passwd'] = self.instance.disk_passwd
         return cdata
 
     def middleware_clean(self, data):
         self.instance.id = self.instance.pk
         data.pop('name')
         data.pop('passwd2', None)
+        data.pop('reset_password', None)
         data.pop('serial')
+
+        sed_passwd = data.get('passwd', '')
+        if self.original_instance['disk_passwd'] != sed_passwd:
+            data['passwd'] = notifier().pwenc_decrypt(sed_passwd)
```

Comment:
How about not sending `passwd` at all if not set by user? That seems safer. pwenc_decrypt may raise exception if user loses the key. Just need to make sure middlewared supports it. `if 'passwd' not in data` or something.